### ISSUE TITLE
CA1801 - false positive for pattern matching

### ISSUE THREAD

**Issue by zvirja** [dotnet/roslyn-analyzers#1326]
#### Analyzer package

Microsoft.CodeAnalysis.FxCopAnalyzers

#### Analyzer

CA1801:Review unused parameters

#### Repro steps

```c#
private static object GetResult(object arg)
{
    switch (arg)
    {
        case string str:
            return str;

        default:
            return null;
    }
}
```

#### Expected behavior
No warning that `arg` is not used

#### Actual behavior
Analyzer highlights that `arg` parameter is not used.

**Comment by zvirja** [comment]
Closing as a duplicate of #1294
